### PR TITLE
Fix bug 1185698: Remove Docker environment.

### PR DESCRIPTION
**Pull request by Osmose**
Removes Docker-specific files and updates the documentation to remove references
to Docker.

@mathjazz r?

### CODE REVIEW

File: docs/dev/install.rst [hunk 1]
```diff
@@ -1,67 +1,108 @@
...
+computer for development.
 
-2. Clone this repository or your fork_:
+Prerequisites
+-------------
+This guide assumes you have already installed and set up the following:
+
+1. Git_
+2. `Python 2.7`_ and pip_
+3. `Node.js`_ and npm_
```
Comment (mathjazz) [inline docs/dev/install.rst]: Where do we need those two?

File: docs/dev/install.rst [hunk 2]
```diff
@@ -1,67 +1,108 @@
...
 
    .. code-block:: bash
 
       git clone --recursive https://github.com/mozilla/pontoon.git
       cd pontoon
 
-3. **Optional**: If you're running the site via boot2docker_, you'll want to add
-   a ``.env`` file to the project's root director with the IP address of the
-   boot2docker VM.
+2. Install the dependencies using peep_, a wrapper around pip that is
```
Comment (mathjazz) [inline docs/dev/install.rst]: Shall we suggest using virtualenv?

**Comment by mathjazz** [comment]
This looks good, but I can't set it up locally. I followed the steps above (with the addition of installing requirements into a virtualenv) and experienced two issues:

Apparently new hashed need to be added to some requirements:
```
THE FOLLOWING PACKAGES DIDN'T MATCH THE HASHES SPECIFIED IN THE REQUIREMENTS
FILE. If you have updated the package versions, update the hashes. If not,
freak out, because someone has tampered with the packages.

    argparse: expected 0Bl46ajFHqezTscfP87RMmQ3zTZrnC4ZJlrOBQkk4OI
                   got s6eaI9N7WgL6pVC5LLu-vrSqHXfmScPrOcGav1Ji2gQ
    polib: expected seoUHVjtXkiu0mdPfIlN-4P2OcMobXsysuGfoDKltAA
                got INKg1YmmksEd9Um9fNqDxmXu8qg-AXuEP-zflW7brXQ
    pytz: expected NYsFoMJgXBXPgzfRcBbrla_qrQfJAN-cLN524hlKklg
               got PhW0FsmiA5waUSCLLNO7T_15bNGeYBsdJlevy3fD3JA
    six: expected QYqTw5en7asj5ViNvAZ6x0pyPts9VBvUk295R252Rdo
              got 4kBSQR_E-9H2cmNVN8P8IzDZSBsYwDF2lbRiWVEskdU
    django-pipeline: expected dFYt2z4rkbygR3Lx_YLP6JEts2H7SuDShiua0EfioVk
                          got ByQht478gXZBLpRL8auUDiO2WLInD0Xw8O-aw61q5gU

-------------------------------
Not proceeding to installation.
```

After I modified requirements, I was able to complete the rest of the steps. But when I tried to access the page in the browser, I got the following error:
```
Environment:


Request Method: GET
Request URL: http://localhost:8000/

Django Version: 1.8.3
Python Version: 2.7.6
Installed Applications:
('pontoon.base',
 'pontoon.administration',
 'pontoon.intro',
 'pontoon.translate',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.staticfiles',
 'commonware.response.cookies',
 'django_browserid',
 'django_jinja',
 'django_nose',
 'pipeline',
 'session_csrf')
Installed Middleware:
('sslify.middleware.SSLifyMiddleware',
 'raygun4py.middleware.django.Provider',
 'django.middleware.common.CommonMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'session_csrf.CsrfMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')


Traceback:
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  132.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/Users/mathjazz/Desktop/pontoon/pontoon/base/views.py" in home
  73.     return translate(request, locale, project.slug)
File "/Users/mathjazz/Desktop/pontoon/pontoon/base/views.py" in translate
  296.     return render(request, template, data)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/shortcuts.py" in render
  67.             template_name, context, request=request, using=using)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/template/loader.py" in render_to_string
  98.             template = get_template(template_name, using=using)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/template/loader.py" in get_template
  29.     engines = _engine_list(using)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/template/loader.py" in _engine_list
  144.     return engines.all() if using is None else [engines[using]]
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/template/utils.py" in all
  110.         return [self[alias] for alias in self]
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/template/utils.py" in __getitem__
  101.             engine = engine_cls(params)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django_jinja/backend.py" in __init__
  86.         self.env = environment_cls(**options)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/jinja2/environment.py" in __init__
  278.         self.extensions = load_extensions(self, extensions)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/jinja2/environment.py" in load_extensions
  75.             extension = import_string(extension)
File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/jinja2/utils.py" in import_string
  213.         return getattr(__import__(module, None, None, [obj]), obj)

Exception Type: ImportError at /
Exception Value: No module named jinja2.ext
```

If I run ```pip freeze```, it gives me:
```
(venv)Snowball:pontoon mathjazz$ pip freeze
Babel==1.3
Django==1.8.3
Jinja2==2.5.5
Pygments==2.0.2
Sphinx==1.3.1
alabaster==0.7.4
argparse==1.3.0
blessings==1.6
blinker==1.3
certifi==14.05.14
chardet==2.3.0
commonware==0.4.3
configparser==3.5.0b2
diff-match-patch==20121119
dj-database-url==0.3.0
django-browserid==1.0.0
django-bulk-update==1.1.4
django-dotenv==1.3.0
django-jinja==1.3.3
django-nose==1.4
django-pipeline==1.5.0
django-session-csrf==0.5
django-sha2==0.4
django-sslify==0.2.7
docutils==0.12
factory-boy==2.5.2
gunicorn==19.3.0
jsonfield==1.0.3
jsonpickle==0.7.0
l20n==0.1
lxml==3.4.4
mercurial==3.4.1
newrelic==2.50.0.39
nose==1.3.6
nose-progressive==1.5.1
oauthlib==0.1.3
parsimonious==0.6.2
polib==1.0.6
psycopg2==2.6
py-bcrypt==0.3
pyasn1==0.1.7
python-Levenshtein==0.11.2
pytz==2015.2
raygun4py==3.0.2
requests==2.6.2
rsa==3.1.4
silme==0.9b
six==1.9.0
snowballstemmer==1.2.0
sphinx-rtd-theme==0.1.8
suds==0.4
test-utils==0.3
translate-toolkit==1.13.0
whitenoise==1.0.6
wsgiref==0.1.2
```

File: docs/dev/install.rst (hunk 1 above)
Comment (Osmose) [inline docs/dev/install.rst]: For compiling the `es6` JavaScript files, although those are only used on the hidden translate pages. Eventually if we switch to es6 for the majority of the site's JS (which I think we should over time) we'll need it.

File: docs/dev/install.rst (hunk 2 above)
Comment (Osmose) [inline docs/dev/install.rst]: I knew I forgot something!

**Comment by Osmose** [comment]
Hmm, I'm unable to replicate either of these. Does anything change if you install via pip instead of peep?

**Comment by mathjazz** [comment]
I guess my problems are not related to this, so r+.

**Pull request merged**